addon/-private/functional/modifier.ts

```diff
@@ -54,14 +60,72 @@ type Teardown = () => unknown;
 export default function modifier<
   E extends Element,
   P extends unknown[],
-  N extends Record<string, unknown>
+  N extends object
```

Comment:
I don't fully grok this change, what does changing to `object` vs `Record<string, unknown>` get us?